### DISCUSSION TITLE
Add OIDC provider to self-hosted Ory Kratos

### DISCUSSION THREAD

**Discussion by spg** [ory/kratos#3415] [Q&A] [answered]
Where can I find Microsoft OIDC documentation for **self-hosted** Kratos? It seems like the only doc I find is for Ory Cloud or using the CLI (I don't use either)

https://www.ory.sh/docs/kratos/social-signin/microsoft

**Comment by vinckr** [comment, marked as answer]
Hello @spg 

You can follow the CLI instructions to add OIDC to self-hosted Ory Kratos. 

So instead of using the CLI to upload the configuration to Ory Network you edit your Ory Kratos config directly, depending on how you deploy it. Also the Jsonnet snippet you host somewhere where your Ory Kratos can read it. 
In general you can use all guides for Ory Network also for self-hosted Ory services, unless it is explicitly mentioned as the API and features are the same.

I hope that makes it clear.
